### PR TITLE
Add "always refresh" option

### PR DESCRIPTION
**Pull request by grahamgilchrist**
Adds a provider option which always cleans the cache when online. This means you can set a very long maxAge on your cache to keep it for offline users, but always refresh with the latest data when online.

### CODE REVIEW

**Comment by grahamgilchrist** [comment]
@incuna/js Please review/merge

**Comment by pandalion** [comment]
LGTM 👍

File: README.md [hunk 1]
```diff
@@ -47,6 +47,16 @@ module.config(function (offlineProvider) {
 });
 ```
 
+### offlineProvider.alwaysRefresh(value)
```
Comment (henrahmagix) [inline README.md:50]: `(value)` => `(boolean)`?

File: README.md [hunk 2]
```diff
@@ -47,6 +47,16 @@ module.config(function (offlineProvider) {
...
+
+Enable or disable always refreshing. When this is `true`, the cache is always refreshed for GET request when online. This enables you to set a very long cache `maxAge`, but still always update the cache when online. Without this, long cache lifetime requests will always be served from the cache, even when online.
```
Comment (henrahmagix) [inline README.md:52]: `long cache lifetime` sounds like a tautology. How about:

```
Without this, requests will always be served from the cache, even when online.
```

Comment (henrahmagix) [inline README.md:52]: `for GET request when online` ->
`for GET requests when online`

File: angular-offline.js [hunk 3]
```diff
@@ -42,6 +42,19 @@ angular
...
   };
 
+
+  /**
+   * Enable or disable always refreshing when online
+   *
+   * @param {boolean} value
+   * @returns {offlineProvider}
+   */
+
```
Comment (henrahmagix) [inline angular-offline.js:52]: Remove blankline between docs and definition?

File: angular-offline.js [hunk 4]
```diff
@@ -66,18 +79,20 @@ angular
...
      * @param {string} key Cache key
      */
 
-    function cleanIfExpired(cache, key) {
+    function clean(cache, key) {
       if (cache === true)
         cache = $requester.defaults.cache || $cacheFactory.get('$http');
       var info = cache.info(key);
-      if (info && info.isExpired)
+
```
Comment (henrahmagix) [inline angular-offline.js:92]: I don't think this blank line is necessary

**Review by henrahmagix — COMMENTED** [review]
👍

**Pull request merged**